lib/css_parser/rule_set.rb

```diff
@@ -512,6 +512,11 @@ def create_background_shorthand! # :nodoc:
     def create_border_shorthand! # :nodoc:
       values = []
 
+      # can't merge if not shortened
+      declarations.each do |name, _value|
+        return nil if %w[border-top border-right border-bottom border-left].any? { |e| name.include?(e) }
+      end
```

Comment:
something like this ?
```suggestion
      return nil if (declarations.keys & BORDER_STYLE_PROPERTIES).any?
```

Reply:
Mh I'm not sure to follow, the idea I had in mind was preventing from merging `BORDER_STYLE_PROPERTIES` together into a new `border` property that would be appended to the declarations, hence overwriting all other properties (even if a border-top/left/bottom/right was initially set after any `BORDER_STYLE_PROPERTIES` )

Your suggestion would abort if there is any `BORDER_STYLE_PROPERTIES` but we're supposed to iterate on them after so it's completely breaking it no ? (Or maybe it was just an example of structure you prefer and what you meant is actually to move the array in a new constant and add a `.keys` method to declarations to get all the declaration names)

Reply:
declarations is a hash, so if I want to check if border-top is defined in that hash I can either do `hash.each { |k,v| k.include? "border-top" }` which was the initial commit
or: `hash.each { |k,v| k == "border-top" }`
or: `hash.key? "border-top"`

also the list of values this was checking looks like BORDER_PROPERTIES, so I suggested replacing that
(but did a mistake and used BORDER_STYLE_PROPERTIES ... so should be a new BORDER_DIRECTIONS constant ?)